### PR TITLE
CassandraQL native support #1355

### PR DESCRIPTION
**Pull request by JiriOndrusek**
fixes https://github.com/apache/camel-quarkus/issues/1355

[ ] An issue should be filed for the change unless this is a trivial change (fixing a typo or similar). One issue should ideally be fixed by not more than one commit and the other way round, each commit should fix just one issue, without pulling in other changes.
[ ] Each commit in the pull request should have a meaningful and properly spelled subject line and body. Copying the title of the associated issue is typically enough. Please include the issue number in the commit message prefixed by #.
[ ] The pull request description should explain what the pull request does, how, and why. If the info is available in the associated issue or some other external document, a link is enough.
[ ] Phrases like Fix #<issueNumber> or Fixes #<issueNumber> will auto-close the named issue upon merging the pull request. Using them is typically a good idea.
[ ] Please run mvn process-resources -Pformat (and amend the changes if necessary) before sending the pull request.
[ ] Contributor guide is your good friend: https://camel.apache.org/camel-quarkus/latest/contributor-guide.html

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: integration-tests/cassandraql/src/main/java/org/apache/camel/quarkus/component/cassandraql/it/CassandraqlResource.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        final Exchange exchange = consumerTemplate
+                .receive(createUrl(String.format("UPDATE employee SET name = '%s', address = '%s' WHERE id = %s",
+                        employee.getName(), employee.getAddress(), employee.getId())));
+        return exchange != null;
+    }
+
+    @Path("/deleteEmployee")
+    @POST
+    @Consumes(MediaType.TEXT_PLAIN)
+    public void updateEmployee(String id) throws Exception {
```
Comment (aldettinger) [inline integration-tests/cassandraql/src/main/java/org/apache/camel/quarkus/component/cassandraql/it/CassandraqlResource.java:90]: What about renaming to something like getEmployeeById ?

Comment (JiriOndrusek) [reply integration-tests/cassandraql/src/main/java/org/apache/camel/quarkus/component/cassandraql/it/CassandraqlResource.java:90]: I see now, that there is a bad method name vs path, which does not make sense, I'll fix it

**Review by aldettinger — APPROVED** [review]
This PR looks good. I find it interesting to leverage on matchers for Employee. Well done :+1:

File: integration-tests/cassandraql/src/main/java/org/apache/camel/quarkus/component/cassandraql/it/CassandraqlResource.java (hunk 1 above)
Comment (JiriOndrusek) [reply integration-tests/cassandraql/src/main/java/org/apache/camel/quarkus/component/cassandraql/it/CassandraqlResource.java]: Correct functionality is to delete by Id, I renamed method to `deleteEmployeeById`.

**Review by jamesnetherton — APPROVED** [review]

**Comment by jamesnetherton** [comment]
The `camel-master` branch needs some tidying so its expected to see some CI failures for docs. I'm working to fix that up.

**Review by zbendhiba — APPROVED** [review]

File: integration-tests/cassandraql/src/test/java/org/apache/camel/quarkus/component/cassandraql/it/CassandraqlTest.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+@QuarkusTestResource(CassandraqlTestResource.class)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+class CassandraqlTest {
+
+    private Employee sheldon = new Employee(1, "Sheldon", "Alpha Centauri");
+    private Employee leonard = new Employee(2, "Leonard", "Earth");
+    private Employee irma = new Employee(3, "Irma", "Jupiter");
+
+    @Test
+    @Order(1)
```
Comment (ppalaga) [inline integration-tests/cassandraql/src/test/java/org/apache/camel/quarkus/component/cassandraql/it/CassandraqlTest.java:45]: Let's have some academic discussion about test structuring ;)

I see that splitting into the methods helps structuring and thus understanding the code.

I see that `@Order` is intended to overcome the fact that the test methods depend on effects caused by their predecessors. This helps when the whole class is executed without any explicit method includes/excludes.

However, `@Order` does not express the dependency. Therefore, when the first test fails, the successor tests are still executed and will most probably fail. They will all be reported as failed and I am not sure if there is any way to see quickly from the report which one of them is the root cause.

Also, the test methods cannot be executed independently via `mvn clean test -Dtest=CassandraqlTest#testUpdate` (except for the first one). 

Given the above, I wonder whether this way of structuring brings enough benefits. I tend to see this kind of situation as end-to-end scenario that is best packed into as single test method where the structure is marked only by comments. (This is not a request to change this PR.) What do others think? Are there any other options to consider?

Comment (jamesnetherton) [reply integration-tests/cassandraql/src/test/java/org/apache/camel/quarkus/component/cassandraql/it/CassandraqlTest.java:45]: Yes, personally I'm not a fan of `@Order`. Like you say, most of our tests follow a pattern when if multiple steps are required to achieve some expected outcome, then they are bundled together in a single test case and marked by comments. E.g:

```
// Create foo

// Retrieve foo

// Delete foo

// Confirm foo was deleted
```

I'd like to 'close down' the `camel-master` branch and get the 3.8 upgrade merged to master soon. So if we can merge this now, we can have a follow up issue to restructure things later if we think its necessary.

**Review by ppalaga — APPROVED** [review]

File: integration-tests/cassandraql/src/test/java/org/apache/camel/quarkus/component/cassandraql/it/CassandraqlTest.java (hunk 2 above)
Comment (ppalaga) [reply integration-tests/cassandraql/src/test/java/org/apache/camel/quarkus/component/cassandraql/it/CassandraqlTest.java:45]: This PR is good to merge. Thanks for your opinion.

**Pull request merged**